Datastructures/queue_template.cpp

```diff
@@ -0,0 +1,104 @@
+#include<iostream>
+using namespace std;
+
+template <class t>
+class MyQueue
+{
+    int s , f , e ;
+    t * arr;
+public:
+    MyQueue()
+    {
+        s=0;
+        f=e=-1;
+    }
+    MyQueue(int n)
+    {
+        s = n;
+        arr = new t[s];
+        f = e = 0;
+    }
+    int Size()
+    {
+        return s;
+    }
+    t Front()
+    {
+        return arr[f];
+    }
+    t Back()
+    {
+        return arr[e];

```

Comment:
before returning Front / Back you should check whether Queue is empty / not.
Back (rear) element should be **arr[e-1]**